platform/mellanox/mlnx-platform-api/sonic_platform/dpuctl_hwm.py

```diff
@@ -0,0 +1,236 @@
+                                            f"dpu{self.index}_shtdn_ready")
+        try:
+            get_shtdn_inotify = InotifyHelper(get_shtdn_ready_path)
+            dpu_shtdn_rdy = get_shtdn_inotify.add_watch(WAIT_FOR_SHTDN, 1)
+        except (FileNotFoundError, PermissionError) as inotify_exc:
+            raise type(inotify_exc)(f"{self.get_name()}:{str(inotify_exc)}")
+        if dpu_shtdn_rdy is None:
+            print(f"{self.get_name()}: Going Down Unsuccessful")
+            self.dpu_power_off(forced=True)
+            self.dpu_power_on(forced=True)
+            return
+
+    def dpu_power_off(self, forced=False):
+        """Per DPU Power off API"""
+        print(f"{self.get_name()}: Power off forced={forced}")
+        if forced:
+            self.write_file(self.set_pwr_f_path, "1")
+        else:
+            self.write_file(self.set_go_down_path, "1")
+            self.dpu_go_down()
+            self.write_file(self.set_pwr_path, "1")
+        print(f"{self.get_name()}: Power Off complete")
+
+    def dpu_power_on(self, forced=False, count=4):
+        """Per DPU Power on API"""
+        if count < 4:
+            print(f"{self.get_name()}: Failed! Retry {4-count}..")
+        print(f"{self.get_name()}: Power on forced={forced}")
+        if forced:
+            self.write_file(self.set_pwr_f_path, "0")
+        else:
+            self.write_file(self.set_pwr_path, "0")
+        get_rdy_inotify = InotifyHelper(self.get_dpu_rdy_path)
+        dpu_rdy = get_rdy_inotify.add_watch(WAIT_FOR_DPU_READY, 1)
+        if not dpu_rdy:
+            if forced:
+                if count > 1:
+                    time.sleep(1)
+                    self.dpu_power_off(forced=True)
+                    self.dpu_power_on(forced=True, count=count-1)
+                else:
+                    print(f"{self.get_name()}: Failed Force power on! Exiting")
+                    return False
+            else:
+                self.dpu_power_off(forced=True)
+                self.dpu_power_on(forced=True)
+        else:
+            print(f"{self.get_name()}: Power on Successful!")
+            return True
+
+    def dpu_reboot_prep(self):
+        """Per DPU Reboot API"""
+        # TODO: Shutdown SONiC on DPU -> SSH Connection -> Shutdown script
+
+    def dpu_burn_fw(self, path):
+        """Per DPU Firmware Update API"""
+        # TODO: Uncomment to install the bfb Image
+        """if not os.path.isfile(path):
+            raise FileNotFoundError(f"{self.get_name()}:File "
+                                    f"{self.file_name} does not exist!")
+        cmd = ["sonic_bfb_install" ,"-b" ,path,"-r","rshim"+str(self.index)]
+        try:
+            cmd_output = subprocess.check_output(cmd)
+        except subprocess.CalledProcessError as cmd_exc:
+            print("Installation failed! code",
+                  cmd_exc.returncode,
+                  cmd_exc.output)"""
+
+    def dpu_pci_scan(self):
```

Comment:
This is not needed. It will be part of the vendor-independent implementation

Reply:
Removed